src/aws/cloudwatch-sink.cpp

```diff
@@ -193,9 +193,9 @@ void CloudWatchSink::cloudwatch_logs_thread()
     existing_log_streams_ = list_existing_log_streams();
     while (is_running_)
     {
-        std::unique_lock<std::mutex> lock(logs_mtx_.get());
         try
         {
+            std::unique_lock<std::mutex> lock(logs_mtx_.get());
```

Comment:
Does this change actually fix some issue, or are you just trying to limit the scope of the lock as much as possible? (because, if I understood correctly the lock is not needed in the `catch` block, which just prints directly to `stderr`?)

Reply:
Limit the scope and if theres a random exception that does happen because this is a wrapper to std::mutex